### PR TITLE
feat(appcomposer): add CodeLens link to Application Composer

### PR DESCRIPTION
**Pull request by JLargent42**
## Problem
Users have reported difficulty finding Application Composer in the Toolkit.
## Solution:
This adds an "Open with Application Composer" link using CodeLens to the start of detected CloudFormation / SAM templates.

This PR also removes the requirement that CFN templates are YAML, as they can also be JSON. `js-yaml` correctly parses JSON, so the code changes are minor.

image

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/toolkit/src/applicationcomposer/codeLensProvider.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            const line = document.lineAt(i)
+            const lineContents = line.text.substring(line.firstNonWhitespaceCharacterIndex)
+            if (lineContents.length > 0 && !lineContents.startsWith('#')) {
+                codeLensLine = i
+                break
+            }
+        }
+        const resourcesLoc = new vscode.Range(codeLensLine, 0, codeLensLine, 0)
+        const codeLens = openTemplateInComposerCommand.build().asCodeLens(resourcesLoc, {
+            title: localize('AWS.applicationComposer.codeLens.title', 'Open with Application Composer'),
```
Comment (bildungsroman) [inline packages/toolkit/src/applicationcomposer/codeLensProvider.ts:36]: As mentioned in the demo call, update to "Visualize with Application Composer"

**Review by bildungsroman — APPROVED** [review]
Just commented with updated wording, but we can ship and follow up as well

File: packages/toolkit/src/shared/cloudformation/cloudformation.ts [hunk 2]
```diff
@@ -19,7 +19,8 @@ export const SERVERLESS_API_TYPE = 'AWS::Serverless::Api' // eslint-disable-line
 export const SERVERLESS_FUNCTION_TYPE = 'AWS::Serverless::Function' // eslint-disable-line @typescript-eslint/naming-convention
 export const LAMBDA_FUNCTION_TYPE = 'AWS::Lambda::Function' // eslint-disable-line @typescript-eslint/naming-convention
 
-export const templateFileGlobPattern = '**/*.{yaml,yml}'
+export const templateFileGlobPattern = '**/*.{yaml,yml,json,template}'
+export const templateFileRegexPattern = /.*\.(yaml|yml|json|template)$/i
```
Comment (justinmk3) [inline packages/toolkit/src/shared/cloudformation/cloudformation.ts:23]: Thanks for analyzing relevant codepaths and making informed decisions about them!

**Review by justinmk3 — APPROVED** [review]

**Comment by JLargent42** [comment]
The issue is seemingly related to `js-yaml` returning an `undefined` object instead of an empty object if a JSON file has no braces. Here's a PR to add null checks so that the error doesn't happen. https://github.com/aws/aws-toolkit-vscode/pull/4464

**Pull request merged**